Pass configuration to source and executor plugins

mszostok opened the pull request:
<!-- Thank you for your contribution. Before you submit the pull request:
1. Follow contributing guidelines, templates, the recommended Git workflow, and any related documentation.
2. Test your changes and attach their results to the pull request.
3. Update the relevant documentation.
-->

## Description

Changes proposed in this pull request:

- Pass configuration to source and executor plugins

## Related issue(s)

Fix #872

mszostok commented on internal/source/dispatcher.go:
```diff
@@ -0,0 +1,88 @@
+package source
```
I just split the Dispatcher service into two:
- `Scheduler` analyzes the provided configuration and based on that schedules plugin sources.
- and `Dispatcher` provides functionality to starts a given plugin, watches for incoming events and calling all notifiers to dispatch received event.
-

mszostok commented on pkg/api/source/grpc_adapter.go:
```diff
@@ -7,14 +7,13 @@ import (
 // Source defines the Botkube source plugin functionality.
 type Source interface {
-	Stream(ctx context.Context) (StreamOutput, error)
+	Stream(ctx context.Context, configs [][]byte) (StreamOutput, error)
```
as the `[][]byte` is used if a few places, I was thinking whether to introduce a dedicated type like:
- `type Configuration []byte` so in func signature: `Stream(ctx context.Context, configs []plugin.Configuration)`
- or - `type Configurations [][]byte` so in func signature: `Stream(ctx context.Context, configs plugin.Configurations)`

WDYT? I would require to stop using the `Request` `Response` on Executor plugin. Is it worth it? On the other hand, it will unify the approach and make the API more self-descriptive.

josefkarasek replied:
Would sending `repeated Config` over network instead of `repeated bytes` also make sense?

mszostok replied:
Not sure if I understand that correctly. In my comment I only mentioned the Go implementation where we can do simply type aliasing, I don't think that is also supported by protobufs.

mszostok replied:
After a f2f discussion:
```
message Config {
   RawYAML bytes = 1
}

repeated Config
```

so in Go: `config.RawYAML`.

josefkarasek reviewed (approved):
lgtm

josefkarasek commented on proto/source.proto:
```diff
@@ -11,6 +9,11 @@ message StreamResponse {
 }
 
+message StreamRequest {
+	// Holds list of config YAML definitions in order specified by user
```
What does this comment mean?

mszostok replied:
It means that the `configs` property holds the list of config in YAML format, I also wanted to express that the order is preserved. 

The Go models are later auto-generated and this comment is also above given property so when you use this struct you know what to expect from it.

mszostok commented:
Hi @josefkarasek! 
Thanks for valuable feedback 🙂 In [this commit](https://github.com/kubeshop/botkube/pull/875/commits/3e9622d327f2f4b22c64e616bbf03bca43ed7f34), I updated the gRPC according to our discussion.
I spot and updated the old plugin repository definition too.
